docs/DeveloperGuide.md

```diff
@@ -1,25 +1,196 @@
 # Developer Guide
 
-## Design & implementation
+## Design 
```

Comment:
It would be good if we can have a class diagram to show the overall design of the system.